src/MiddlewareListener.php

```diff
@@ -0,0 +1,77 @@
+<?php
+
+namespace DI\Bridge\Silex;
+
+use DI\Bridge\Silex\Application;
+use Symfony\Component\HttpKernel\Event\GetResponseEvent;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * Replacement for the Silex MiddlewareListener to allow arbitrary injection into middleware functions.
+ *
+ * @author Felix Becker <f.becker@outlook.com>
+ */
+class MiddlewareListener extends \Silex\EventListener\MiddlewareListener
```

Comment:
There are two types of middleware, Application level middleware via `$app->before()` and route / controller collection level middleware like `$app->get('/', ...)->before(...)`. Route level middleware is handled throught `MiddlewareListener`.